common/main/java/com/couchbase/lite/internal/core/C4Listener.java

```diff
@@ -0,0 +1,211 @@
+
+//
+// Copyright (c) 2020 Couchbase, Inc All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+package com.couchbase.lite.internal.core;
+
+import android.support.annotation.GuardedBy;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+import com.couchbase.lite.ConnectionStatus;
+import com.couchbase.lite.LiteCoreException;
+import com.couchbase.lite.internal.utils.MathUtils;
+
+// !!! Temporary warning suppression.  Remove this.
+@SuppressFBWarnings("DLS_DEAD_LOCAL_STORE")
+@SuppressWarnings({"PMD.PrematureDeclaration", "PMD.UnusedLocalVariable", "PMD.UnusedPrivateField"})
+public final class C4Listener extends C4NativePeer {
+    public enum Option {REST, SYNC}
+
+    public enum KeyMode {CERT, KEY}
+
+    private static final Map<Option, Integer> OPTION_TO_C4;
+
+    static {
+        final Map<Option, Integer> m = new HashMap<>();
+        m.put(Option.REST, 0x01);
+        m.put(Option.SYNC, 0x02);
+        OPTION_TO_C4 = Collections.unmodifiableMap(m);
+    }
+
+    private static final Map<KeyMode, Integer> KEY_MODE_TO_C4;
+
+    static {
+        final Map<KeyMode, Integer> m = new HashMap<>();
+        m.put(KeyMode.CERT, 1);
```

Comment:
For consistency, use hex like OPTION_TO_C4 or the other way around?

Reply:
Oh... ok.  Sure.

Reply:
Wait.  Options are bit flags.  They need to be hex.  Certs modes are not.  They are mutually exclusive.

These declarations are identical to the ones in c4Listener.cc